helperFunctions/subscribeHelpers/SalesforceAgentAssignedHelpers/SubscribeToLiveAgentHelper.ts

```diff
@@ -69,24 +73,44 @@ export class SubscribeToLiveAgent {
 						if (chatEndedReason === 'agent') {
 							await updateRoomCustomFields(this.data.room.id, { agentEndedChat: true }, this.read, this.modify);
 							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_BY_AGENT));
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_AGENT);
+						} else {
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_VISITOR);
 						}
-						await handleEndChatCallback.handleEndChat();
+					} else if (isChatRequestFail === true) {
+						const {
+							message: { reason: chatEndedReason },
+						} = getForEvent(messageArray, 'ChatRequestFail');
+						if (chatEndedReason === 'Unavailable') {
+							await updateRoomCustomFields(this.data.room.id, { agentDisconnected: true }, this.read, this.modify);
+							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_LIVEAGENT_DISCONNECTED));
```

Comment:
At other places, we are sending `ESCALATION_FAILED_DUE_TO_NO_LIVEAGENT_AVAILABLE` when we receive an `Unavailable` message. here we are sending `CHAT_CLOSED_LIVEAGENT_DISCONNECTED`